Added Follow/Unfollow feature to Chirp

KampJN opened the pull request.

KampJN committed:
Moved cheep text to a new line

Co-Authored-By: StineHelena <179689613+StineHelena@users.noreply.github.com>

KampJN committed:
Started IFollowService og FollowService

Co-Authored-By: StineHelena <179689613+StineHelena@users.noreply.github.com>

KampJN committed:
Made constructor for FollowService

Co-Authored-By: StineHelena <179689613+StineHelena@users.noreply.github.com>

StineHelena committed:
Set up
Co-Authored-by: Josefine <kajn@itu.dk>

KampJN committed:
Added buttons for Follow/Unfollow

Co-Authored-By: StineHelena <179689613+StineHelena@users.noreply.github.com>

KampJN committed:
Tried adding functions to the buttons

Co-Authored-By: StineHelena <179689613+StineHelena@users.noreply.github.com>

KampJN committed:
Added missing semicolon

KampJN committed:
Merge branch 'primary' into FollowMe

KampJN committed:
The buttons work?

The page can now run, but it crashes when the user tries to click on an author. The follow/unfollow have not been tested

Co-Authored-By: StineHelena <179689613+StineHelena@users.noreply.github.com>

KampJN committed:
The page loads, but the buttons do not work

Co-Authored-By: StineHelena <179689613+StineHelena@users.noreply.github.com>

KampJN committed:
Made a function for getCheepsFromFollowing

KampJN committed:
Made a function to check is a user is in _following

KampJN committed:
Corrections

KampJN committed:
Made bool to check for following

KampJN committed:
Fixing errors

KampJN committed:
Fixed IFollowService i Program.cs

Co-Authored-By: Amira Maria Ayoub <151861445+milkboy23@users.noreply.github.com>

KampJN committed:
Corrections and added if statement for buttons

KampJN committed:
Fixed if statement for buttons

KampJN committed:
Corrections

Now working on a "value cannot be null" error when clicking Follow

KampJN committed:
Fixed a misstype

KampJN committed:
Now it doesn't crash

KampJN committed:
Corrections

KampJN committed:
Added follower relation to the db

KampJN committed:
Added functions to AuthorRepository

KampJN committed:
Made connection between repo and html

KampJN committed:
A user can now follow, no cheeps appear

KampJN committed:
Users can now follow other users

KampJN committed:
Added co-author comments

I wrote the last few commits with someone from the university, that is not in our group. We forgot to co-author him on the commits, so now I've added comments with his name and email, as well as put him as co-author on this commit

EmmaAndrea commented:
Good work Jose! 
Team: Remember to delete your local database, and remove and remake your migrations:))

KampJN committed:
Fixed a bug

A user could not see cheeps on a user timeline unless they were logged in. Now they can see the cheeps, but can only follow if logged in

Co-Authored-By: Matt Gauger <matt.gauger@gmail.com>
Co-Authored-By: StineHelena <179689613+StineHelena@users.noreply.github.com>
